test/integration/transaction_list_ft_tests.js

```diff
@@ -594,6 +594,98 @@ describe('The /transactions endpoint filtering by states)', () => {
       })
       .end(done)
   })
+
+  it('should return a list of transactions with corporate surcharge total amount when relevant', function (done) {
```

Comment:
It's great that you added tests for the change here 👍 

Currently we are expecting as a result JSON, which is not ideal and it is not fully tested (in terms of integration test as it skips the view), if you can see how we set the `supertest` - https://github.com/alphagov/pay-selfservice/blob/afcd4dcf8602049f6c591bf9d517265e823ffd65/test/integration/transaction_list_ft_tests.js#L609

As discussed before, we are trying to avoid using this approach and move to `Cypress` or `supertest` but not to `assert` on JSON.

We have Cypress test for the page already here:

- https://github.com/alphagov/pay-selfservice/blob/afcd4dcf8602049f6c591bf9d517265e823ffd65/test/cypress/integration/user/transaction_search_spec.js#L14

On the other option `supertest` + `cheerio` you can have a look as an example here https://github.com/alphagov/pay-selfservice/blob/afcd4dcf8602049f6c591bf9d517265e823ffd65/test/unit/controller/edit_merchant_details_controller/get-index.test.js#L65

I think `Cypress` is better to have as a test as it tests real interaction.

In case you think `Cypress` is good option you can change https://github.com/alphagov/pay-selfservice/blob/afcd4dcf8602049f6c591bf9d517265e823ffd65/test/cypress/integration/user/transaction_search_spec.js file and add for example following:

```
  const selfServiceDefaultUser = selfServiceUsers.config.users.filter(fil => fil.isPrimary === 'true')[0]

  const aCorporateCardSurchargeCharge = selfServiceDefaultUser.sections.transactions.data.filter(fil => fil.corporate_card_surcharge !== undefined)[0]
```

to https://github.com/alphagov/pay-selfservice/blob/afcd4dcf8602049f6c591bf9d517265e823ffd65/test/cypress/integration/user/transaction_search_spec.js#L6

and later add a test, initial implementation can be like:

```JS
    it('should display card fee with corporate surcharge transaction', () => {
      // Ensure the transactions list has the right number of items
      const unfilteredTransactions = selfServiceDefaultUser.sections.transactions.data
      cy.get('#transactions-list tbody').find('tr').should('have.length', unfilteredTransactions.length)

      // Ensure the values are displayed correctly
      cy.get('#transactions-list tbody').find('tr').eq(4).find('td').eq(1).should('contain', convertAmounts(aCorporateCardSurchargeCharge.total_amount)).and('contain', '(with card fee)')
    })
```

Reply:
As discussed I have paired with @SandorArpa to add the Cypress test